plugins/catalog-react/src/components/EntityRefLink/EntityRefLink.tsx

```diff
@@ -39,6 +61,104 @@ export type EntityRefLinkProps = {
   children?: React.ReactNode;
 } & Omit<LinkProps, 'to'>;
 
+type PeekAheadPopoverProps = {
+  popupState: PopupState;
+  entityRef: CompoundEntityRef;
+};
+
+const useStyles = makeStyles(() => {
+  return {
+    popoverPaper: {
+      width: '20em',
+    },
+  };
+});
+
+export const PeekAheadPopover = ({
+  popupState,
+  entityRef,
+}: PeekAheadPopoverProps) => {
+  const entityRoute = useRouteRef(entityRouteRef);
+  const classes = useStyles();
+  const catalogApi = useApi(catalogApiRef);
+
+  const {
+    value: entity,
+    loading,
+    error,
+  } = useAsync(async () => {
+    if (popupState.isOpen) {
+      const retrievedEntity = await catalogApi.getEntityByRef(entityRef);
+      if (!retrievedEntity) {
+        throw new Error(`${entityRef.name} was not found`);
+      }
+      return retrievedEntity;
+    }
+    return undefined;
+  }, [popupState]);
+
+  if (loading) {
+    return null;
+  }
+
+  return (
+    <HoverPopover
+      PaperProps={{
+        className: classes.popoverPaper,
+      }}
+      {...bindPopover(popupState)}
+      anchorOrigin={{
+        vertical: 'bottom',
+        horizontal: 'center',
+      }}
+      transformOrigin={{
+        vertical: 'top',
+        horizontal: 'center',
+      }}
+    >
+      <Card>
+        <CardContent>
+          <Typography gutterBottom>{entityRef.namespace}</Typography>
```

Comment:
The spacings were a bit odd in the screenshot, the card docs have some suggested styles to draw inspiration from https://v4.mui.com/components/cards/

Reply:
I put a screenshot above I think it looks a little better. Let me know what you think.